singularity/Singularity

```diff
@@ -85,13 +85,11 @@ gpgkey=https://www.mongodb.org/static/pgp/server-3.4.asc" >> /etc/yum.repos.d/mo
     ## clear all caches
     yum clean all
     conda clean -y --all
-    npm cache clean --force
-    rm -rf /tmp/* /var/tmp/*
```

Comment:
Cleaning cache before finalizing containers is the standard practice. It helps to release some space. Bring them back unless there is a reason for not doing that.

Reply:
I removed them because they broke the build. `npm cache clean` is not recommended anymore because npm automatically clears whatever cached data needs to be cleared. The second line made the built fail as well:

```
rm: cannot remove '/tmp/build-temp-563787328/rootfs/dev': Device or resource busy
rm: cannot remove '/tmp/build-temp-563787328/rootfs/etc/hosts': Device or resource busy
rm: cannot remove '/tmp/build-temp-563787328/rootfs/etc/resolv.conf': Device or resource busy
rm: cannot remove '/tmp/build-temp-563787328/rootfs/proc': Device or resource busy
rm: cannot remove '/tmp/build-temp-563787328/rootfs/sys': Device or resource busy
rm: cannot remove '/tmp/build-temp-563787328/rootfs/tmp': Device or resource busy
rm: cannot remove '/tmp/build-temp-563787328/rootfs/var/tmp': Device or resource busy
FATAL:   While performing build: while running engine: exit status 1
```

Reply:
Okay did you set `SINGULARITY_TMPDIR`?

Reply:
My image is `619M` while the new one is `819M`. I wonder if it is due to embodying `/tmp/` or just node version change. What do you think?

Reply:
You haven't lost much by not cleaning them. The following will not mount `/tmp` and `/var/tmp` and lets us examine the space consumed during building process:

> singularity shell --no-mount tmp dpdash-ben.sif

They are nearly zero.

Reply:
You're right. So it should be OK as is, then?

Reply:
Yes